src/components/ErrorModal.vue

```diff
@@ -0,0 +1,264 @@
+<template>
+  <div
+    class="error-modal"
+    @click="resolve"
+  >
+    <div class="modal-body" @click.stop>
+      <IconClose
+        class="close-modal"
+        @click="resolve"
+      />
+      <template v-if="!isSent">
+        <FailureIcon />
+        <h1>{{ $t('reportBug.Title') }}</h1>
+        <h2>{{ $t('reportBug.SubTitle') }}</h2>
+        <p class="description">
+          {{ $t('reportBug.Description') }}
+        </p>
+        <ButtonPlain
+          class="details-button"
+          @click="toggleDetails"
+        >
+          <IconEye />
+          {{ $t('reportBug.ViewDetails') }}
+        </ButtonPlain>
+        <p v-show="details">
+          {{ JSON.stringify(error) }}
+        </p>
+        <h2 class="report-title">
+          {{ $t('reportBug.DescriptionTitle') }}
+        </h2>
+        <textarea
+          v-model="report"
+          class="report-text"
+          :placeholder="$t('reportBug.Placeholder')"
+        />
+        <Button
+          class="cancel"
+          @click="resolve"
+        >
+          {{ $t('reportBug.Cancel') }}
+        </Button>
+        <Button
+          @click="send"
+        >
+          {{ $t('reportBug.Send') }}
+        </Button>
+      </template>
+      <template v-else>
+        <IconOK />
+        <h1 class="success-title">
+          {{ $t('reportBug.Success.Title') }}
+        </h1>
+        <h2 class="success-subtitle">
+          {{ $t('reportBug.Success.SubTitle') }}
+        </h2>
+        <p class="success-description">
+          {{ $t('reportBug.Success.Description') }}
+        </p>
+        <Button
+          @click="resolve"
+        >
+          {{ $t('reportBug.Success.Return') }}
+        </Button>
+      </template>
+    </div>
+  </div>
+</template>
+
+<script>
+import { detect } from 'detect-browser';
+import FailureIcon from '../assets/APIError.svg?icon-component';
+import IconClose from '../assets/iconCloseRebranded.svg?icon-component';
+import IconEye from '../assets/iconEye.svg?icon-component';
+import IconOK from '../assets/iconOK.svg?icon-component';
+import ButtonPlain from './ButtonPlain.vue';
+import Backend from '../utils/backend';
+import Button from './Button.vue';
+
+export default {
+  components: {
+    FailureIcon,
+    IconClose,
+    IconEye,
+    IconOK,
+    ButtonPlain,
+    Button,
+  },
+  props: {
+    resolve: { type: Function, required: true },
+    error: { type: [Object, Error], required: true },
+  },
+  data() {
+    return {
+      report: '',
+      isSent: false,
+      details: false,
+    };
+  },
+  methods: {
+    async send() {
+      const report = {
+        appVersion: process.env.npm_package_version,
+        browser: detect(),
+        error: this.error,
+        platform: 'superhero-ui',
+        description: this.report,
+        time: Date.now(),
```

Comment:
maybe more useful to remember the time when the modal is opened to don't record how long the user need to fill the description and to make easily to find the corresponding request to the backend (in case if error popup arrear because of backend response)

Reply:
@davidyuk I will move `Date.now()` outside of the Vue component, and add it to the plugin.